File: testers/db2/java/src/java/testdb/Db2TestHelper.java
package testdb;

import dev.typr.foundations.Connection;
import dev.typr.foundations.ConnectionRead;
import dev.typr.foundations.SqlConsumer;
import dev.typr.foundations.SqlFunction;
import dev.typr.foundations.Transactor;
import dev.typr.foundations.connect.Db2Config;

public class Db2TestHelper {
  private static final Db2Config CONFIG =
      Db2Config.builder("localhost", 50000, "typr", "db2inst1", "password").build();

  private static final Transactor TRANSACTOR = Transactor.create(CONFIG).rollbackOnly();

  public static <T> T apply(SqlFunction<Connection, T> f) {
    return TRANSACTOR.transact(f);
  }

  public static void run(SqlConsumer<Connection> f) {
    TRANSACTOR.transactVoid(f);
  }

  public static <T> T applyRead(SqlFunction<ConnectionRead, T> f) {
    return TRANSACTOR.transactRead(f);
  }
}

File: testers/db2/kotlin/src/kotlin/testdb/Db2TestHelper.kt
package testdb

import dev.typr.foundationskt.Connection
import dev.typr.foundationskt.ConnectionRead
import dev.typr.foundationskt.Transactor
import dev.typr.foundationskt.connect.Db2Config

object Db2TestHelper {
    private val CONFIG: Db2Config =
        Db2Config.builder("localhost", 50000, "typr", "db2inst1", "password").build()

    private val TRANSACTOR: Transactor = Transactor.create(CONFIG).rollbackOnly()

    fun <T> apply(f: (Connection) -> T): T = TRANSACTOR.transact(f)

    fun run(f: (Connection) -> Unit) = TRANSACTOR.transact { conn -> f(conn) }

    fun <T> applyRead(f: (ConnectionRead) -> T): T = TRANSACTOR.transactRead(f)
}
